labgrid/remote/client.py

```diff
@@ -1958,6 +1958,10 @@ def main():
             else:
                 print(f"{parser.prog}: error: {e}", file=sys.stderr)
             print("This may be caused by disconnected exporter or wrong match entries.\nYou can use the 'show' command to review all matching resources.", file=sys.stderr)  # pylint: disable=line-too-long
+            if e.found:
+                print("Found multiple resources but no name was given, available names:")
+                for res in e.found:
+                    print(f"{res.name}")
```

Comment:
I think it would be better to implement this in `__str__()` of `NoResourceFoundError`. Suggestion:
```python
def __str__(self):
    if self.found:
        names = [res.name for res in self.found]
        return f"{', '.join(self.args)}; available names: {', '.join(names)}"

    return self.message
```
Output looks like this:
```
$ labgrid-client --place myplace console
labgrid-client: error: multiple resources matching <class 'labgrid.resource.serialport.NetworkSerialPort'> found in Target(name='myplace', env=None); available names: console, acm
This may be caused by disconnected exporter or wrong match entries.
You can use the 'show' command to review all matching resources.
```
Maybe we could show the class name there instead of its whole representation for even better readability?

Reply:
I think there is a case to be made for both. As a library user descriptive error cases are helpful and including the names definitely helps there, but for client usage I really want to enumerate the individual names on different lines since it makes it a lot more explicit and user friendly.

Reply:
Okay, agreed. Is the "This may be caused by disconnected exporter" message needed in that case or should that happen in an `else` path? This kind of error generates seems to generate a whole mixture of helpful and not so helpful messages. Maybe we can drop some for the multiple resources case?

Reply:
Good point, we should do that since it wasn't a connection error.